src/chart/line/LineView.ts

```diff
@@ -233,17 +238,17 @@ function getVisualGradient(
     // LinearGradient to render `outerColors`.
 
     const axis = coordSys.getAxis(coordDim);
+    const axisScaleExtent = axis.scale.getExtent();
 
-    interface ColorStop {
-        offset: number
-        coord?: number
-        color: ColorString
-    }
     // dataToCoord mapping may not be linear, but must be monotonic.
     const colorStops: ColorStop[] = zrUtil.map(visualMeta.stops, function (stop) {
+        let coord = axis.toGlobalCoord(axis.dataToCoord(stop.value));
+        // normalize the infinite value
+        isNaN(coord) || isFinite(coord)
```

Comment:
We knew Infinite coord is calculated from an infinite value. So can we simplify the code to only clamp the infinite value.

```ts
const value = stop.value;
const coord = axis.toGlobalCoord(axis.dataToCoord(
  value, !(isFinite(value) || isNaN(value))
));
```

Reply:
This is simple and can solve the visual map bug but it can't work for the first case that uses the log axis.

Reply:
Take the first log axis case, it should be caused by `dataToCoord` will probably return an infinite value.

After invoking https://github.com/apache/echarts/blob/4db2f1e7c1eb50cf9305370aa85381e2840623ef/src/coord/Axis.ts#L125
the data may become `Infinity` and we expect `clamp` is `true` but before this, the `clamp` was calculated as `false`.

So it may be better to check again and clamp after the first coordinate conversion.

Now the logic should be like this,

```ts
const value = stop.value || 0;
let coord = axis.toGlobalCoord(axis.dataToCoord(value, !isFinite(value)));
// if coord is still infinite
isNaN(coord) || isFinite(coord) || (coord = axis.toGlobalCoord(axis.dataToCoord(coord, true)));
```

Reply:
Take the first log axis case, it should be because `dataToCoord` will probably return an infinite value.

After invoking https://github.com/apache/echarts/blob/4db2f1e7c1eb50cf9305370aa85381e2840623ef/src/coord/Axis.ts#L125
the data may become `Infinity`.
We expect `clamp` is `true` but before this, `clamp` was calculated as `false`.

So it may be better to check again and clamp after the first coordinate conversion
or modify the `dataToCoord` function (I'm afraid we can't)

Reply:
Understand, thanks for the explaination